pkg/query/querier.go

```diff
@@ -180,7 +189,16 @@ func (q *querier) Select(params *storage.SelectParams, ms ...*labels.Matcher) (s
 
 	queryAggrs, resAggr := aggrsFromFunc(params.Func)
 
-	resp := &seriesServer{ctx: ctx}
+	// We're limiting store read time here.
+	// This limit affects behaviour what result Thanos Query will return to user if one of requested stores timed out.
+	// If query.timeout > store.read-timeout
+	// client will get partial response from stores who responded faster than store.read-timeout
+	// If query.timeout <= store.read-timeout
+	// client will get an error with timeout for whole request even if some of stores responded in time, but one timed out
+	storeReadCtx, storeReadCancelFunc := context.WithTimeout(ctx, q.storeReadTimeout)
+	defer storeReadCancelFunc()
+
+	resp := &seriesServer{ctx: storeReadCtx}
```

Comment:
I think this might be bit wrong. Essentially this timeout, especially if we define it like this should affect only too slow store API. So it means that for use `storeReadTimeout` is essentially `grpc client` timeout, and we should set it in `proxy.go` instead. 

If we do it here we actually have timeout per whole proxy operation.. including all stores as if one fail the whole thing fails.

Reply:
Okay, thanks, I've moved timeout to ProxyStore.